tests/components/fastdotcom/test_coordinator.py

```diff
@@ -0,0 +1,40 @@
+"""Test the FastdotcomDataUpdateCoordindator."""
+from datetime import timedelta
+from unittest.mock import patch
+
+from freezegun.api import FrozenDateTimeFactory
+
+from homeassistant.components.fastdotcom.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import UpdateFailed
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+
+async def test_fastdotcom_data_update_coordinator(
+    hass: HomeAssistant, freezer: FrozenDateTimeFactory
+) -> None:
+    """Test the update coordinator."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id="UNIQUE_TEST_ID",
+    )
+    config_entry.add_to_hass(hass)
+
+    await hass.config_entries.async_setup(config_entry.entry_id)
```

Comment:
patch the library from here and assert the state of the entity.
Then a second test if update fails that assert entity becomes unavailable.